src/test/java/spoon/test/trycatch/TryCatchTest.java

```diff
@@ -442,4 +444,60 @@ public void testTryWithVariableAsResource() {
 		tryStmt.removeResource(ctResource);
 
 	}
+
+	@Nested
+	class AddCatcherAt {
+		@Test
+		void addsCatcherAtTheSpecifiedPosition() {
```

Comment:
Note: This test really tests three separate cases in one go:

1. Inserting into an empty list.
2. Inserting at the beginning of a non-empty list.
3. Inserting at a position that is neither first nor last.

Given that this functionality is neither complicated nor critical to Spoon, that's probably fine. But it always bears thinking about how much is tested in any single test. The fact that we never insert at the end of a non-empty list also isn't a big deal considering what the implementation looks like.

Reply:
> But it always bears thinking about how much is tested in any single test.

Right. Ideally, we should test only one case per _unit_ test. I will keep this in mind. We can skip refactoring for this feature because, like you said, it is not that critical and is rather a very simple feature.

> The fact that we never insert at the end of a non-empty list also isn't a big deal considering what the implementation looks like.

Tests of `addCatcher` would already test that so we do not need to think about them I guess.

Reply:
> Tests of addCatcher would already test that so we do not need to think about them I guess.

Precisely, "considering what the implementation looks like" :). White box testing has merits in that we can omit redundant testing, but it has drawbacks to match (e.g. creating tests based on the implementation rather than the specification).